lib/active_model_serializers/adapter/json_api/deserialization.rb

```diff
@@ -140,12 +140,18 @@ def validate_payload(payload)
         # rubocop:enable Metrics/CyclomaticComplexity
 
         # @api private
-        def filter_fields(fields, options)
+        def filter_fields(hash, options)
+          hash = transform_keys(hash, options)
```

Comment:
if we're to transform the keys for attributes and relationships, then it should happen before this method is called, rather than within it